internal/gitstream/assign.go

```diff
@@ -50,6 +50,62 @@ func (a *Assign) Run(ctx context.Context) error {
 	return nil
 }
 
+func (a *Assign) filterApproversFromCommitAuthors(commitAuthors []string, owners *owners.Owners) []string {
+
+	filteredCommitAuthors := make([]string, 0, len(commitAuthors))
+	for _, ca := range commitAuthors {
+		if a.OwnersHelper.IsApprover(owners, ca) {
+			filteredCommitAuthors = append(filteredCommitAuthors, ca)
+		}
+	}
+
+	return filteredCommitAuthors
+}
+
+func (a *Assign) handleIssue(ctx context.Context, issue *github.Issue, owners *owners.Owners) error {
+
+	logger := a.Logger.WithValues("url", *issue.HTMLURL, "issue", *issue.Number)
+
+	if len(issue.Assignees) > 0 {
+		return nil
+	}
+
+	logger.Info("Processing issue")
+
+	shas, err := a.Finder.FindSHAs(*issue.Body)
+	if err != nil {
+		return fmt.Errorf("error while looking for SHAs in %q: %v", *issue.Body, err)
+	}
+
+	commitAuthors := make([]string, 0, len(shas))
+	for _, s := range shas {
+		user, err := a.UserHelper.GetCommitAuthor(ctx, s.String())
+		if err != nil {
+			return fmt.Errorf("failed to get commit author from GitHub for issue %d in commit %s: %v",
+				*issue.Number, s.String(), err)
+		}
+		commitAuthors = append(commitAuthors, *user.Login)
+	}
+
+	filteredCommitAuthors := a.filterApproversFromCommitAuthors(commitAuthors, owners)
+
+	if len(filteredCommitAuthors) == 0 {
+		randAssignee, err := a.OwnersHelper.GetRandomApprover(owners)
+		if err != nil {
+			return fmt.Errorf("could not get a random approver: %v", err)
+		}
+		filteredCommitAuthors = append(filteredCommitAuthors, randAssignee)
+	}
+
+	for _, userLogin := range filteredCommitAuthors {
+		if err := a.IssueHelper.Assign(ctx, issue, userLogin); err != nil {
```

Comment:
The GitHub client supports adding multiple assignees at once. Let us modify `a.IssueHelper.Assign` to accept multiple usernames via variadic arguments to avoid multiple calls in a loop here.

Reply:
This is a good point. Fixed as well.